server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -157,7 +157,9 @@ public synchronized void updateSegments(final SegmentInfos infos) throws IOExcep
             // Commit and roll the translog when we receive a different generation than what was last received.
             // lower/higher gens are possible from a new primary that was just elected.
             if (incomingGeneration != lastReceivedGen) {
-                commitSegmentInfos();
+                if (engineConfig.getIndexSettings().isRemoteStoreEnabled() == false) {
```

Comment:
rather than having this condition in two spots, can you put it in `commitSegmentInfos` ?

Reply:
It's using two different methods and we'd still have to gate the logic to increment latestSegmentInfos' counter. Think it's cleaner to leave it this way.